devito/nodes.py

```diff
@@ -473,9 +474,19 @@ def __init__(self, name, body, retval, parameters=None, prefix=('static', 'inlin
         self.name = name
         self.body = as_tuple(body)
         self.retval = retval
-        self.parameters = as_tuple(parameters)
         self.prefix = prefix
 
+        if all([isinstance(x, RuntimeArgProvider) for x in parameters]):
+            args = flatten([x.rtargs for x in parameters])
+        else:
+            assert(all([isinstance(x, RuntimeArgument) for x in parameters]))
+            args = parameters
+
+        self.parameters = as_tuple(args)
+
+        self.t_args = [x for x in args if isinstance(x, TensorArgument)]
```

Comment:
I argue that the current codebase uses object attributes in exactly the same way as I have used here, in multiple places. I have included links to some of them but there are many more instances. That should settle your convention argument since this style is currently in use. The other arguments should apply equally to all these instances I list below. Overall, I think this is just bike-shedding. This does not make a difference either way.

https://github.com/opesci/devito/blob/master/devito/nodes.py#L184

https://github.com/opesci/devito/blob/master/devito/dse/clusterizer.py#L21

https://github.com/opesci/devito/blob/master/devito/dse/graph.py#L137

https://github.com/opesci/devito/blob/master/devito/dse/graph.py#L143

https://github.com/opesci/devito/blob/master/devito/dse/graph.py#L148

https://github.com/opesci/devito/blob/master/devito/dle/backends/yask.py#L106

https://github.com/opesci/devito/blob/master/devito/operator.py#L81

https://github.com/opesci/devito/blob/master/devito/operator.py#L557